net/http/upstream.go

```diff
@@ -0,0 +1,90 @@
+package http
+
+import "sync/atomic"
+
+type upstream struct {
+	// Current backend index
+	index uint32
+
+	// Balancing parameters
+	currentWeight int32
+	maxWeight     int32
+
+	// preorety backend
+	preoretyBackend *backend
+
+	// List of the upstream backends
+	backends backends
+
+	// greatest common divisor
+	gcd int32
+}
+
+func (ups *upstream) nextBackend(maxRequestsByBackend int) (back *backend) {
+	// First send requests to the priority backend (generally this is the local service)
+	if ups.preoretyBackend != nil {
+		if maxRequestsByBackend <= 0 || maxRequestsByBackend > ups.preoretyBackend.requestCount() {
```

Comment:
Could you rename `request` to `concurrentRequest` to make the code a little bit more easier for a reader? :)

Reply:
Done